index.js

```diff
@@ -86,7 +87,11 @@ Plugin.prototype.writeOutput = function(compiler, contents) {
     contents.publicPath = compiler.options.output.publicPath;
   }
   mkdirp.sync(path.dirname(outputFilename));
-  fs.writeFileSync(outputFilename, JSON.stringify(contents, null, this.options.indent));
+  if (this.options.append) {
+    fs.appendFileSync(outputFilename, JSON.stringify(contents, null, this.options.indent));
```

Comment:
`appendFileSync` won't work as the file won't be JSON parseable anymore.

What you want is to 

1. `readFileSync()`
2. `JSON.parse()` the content of the file (or default to `{}` if the file is empty)
3. `Object.assign(resultFromParsing, contents);`
4. Stringify + write the output